tools/Dashboard/DevHome.Dashboard/Views/DashboardView.xaml.cs

```diff
@@ -839,8 +863,70 @@ private async void WidgetControl_Drop(object sender, DragEventArgs e)
         }
 
         ViewModel.PinnedWidgets.CollectionChanged += OnPinnedWidgetsCollectionChangedAsync;
+    }
 
-        _log.Debug($"Drop ended");
+    /// <summary>
+    /// Handle keyboard shortcuts for moving widgets left and right.
+    /// </summary>
+    private async void HandleKeyUpAsync(object sender, KeyRoutedEventArgs e)
+    {
+        _log.Debug($"Key up");
+
+        await _moveWidgetsLock.WaitAsync();

```

Comment:
Oh I see, is this lock to prevent concurrent execution from manipulating shared resources? If yes, putting `[RelayCommand]` is sufficient as by default `AllowConcurrentExecutions` is set to false.

References: [RelayCommand](https://learn.microsoft.com/en-us/dotnet/communitytoolkit/mvvm/generators/relaycommand) docs